sync_package_analysis_documents

bissenbay opened the pull request:
Created model for package-analysis documents in dgraph

bissenbay commented on thoth/storages/graph/models.py:
```diff
@@ -52,6 +52,21 @@ class PackageExtractRun(VertexBase):
+    package_analysis_datetime = model_property(type=datetime, index="hour")
+    package_analysis_version = model_property(type=str, index="exact")
+    package_analysis_name = model_property(type=str, index="exact")
+    environment_type = model_property(type=str, index="exact")
```
@fridex Do we need environment_type? What else we would need to store for package analysis documents?

fridex replied:
We will need to first finish the initial design of the package-extract output. Then, take a look at [schema](https://raw.githubusercontent.com/thoth-station/storages/master/docs/schema.png) and try to come up with some changes. We can discuss it in a meeting.

bissenbay replied:
Got it!

fridex replied:
:+1:

bissenbay commented on thoth/storages/graph/models.py:
```diff
@@ -52,6 +52,21 @@ class PackageExtractRun(VertexBase):
 
 
+@attr.s(slots=True)
+class PackageAnalyzerRun(VertexBase):
```
@fridex 1. This is how a new node is created in DGraph, right? Just curious why it's parent class is called VertexBase? 2. Should I just change the name to ```PythonFileDigests``` instead of ```PackageExtractRun``` to sync the results of Package Analyzer?

fridex replied:
`PythonFileDigests` will point to `PythonFileDigests` so keep this one and create `PythonFileDigests`. The base class implements the actual syncing logic, these are leaf classes which define the model properties.

bissenbay replied:
Done

bissenbay commented:
- Created a new model for PythonFileDigests node
- Created a new model for FileFound node
- Python files found in an image are connected to PythonFileDigests node
- Updated schema.rdf
- Deleted model for PackageAnalyzerRun node and corresponding methods to it

fridex commented on thoth/storages/graph/models.py:
```diff
@@ -52,6 +52,20 @@ class PackageExtractRun(VertexBase):
+class FileFound(ReverseEdgeBase):
+    """An edge representing a filepath to the python file found in an image."""
+
+    filepath = model_property(type=str, index="exact", default="*")
```
No default as its always set.

bissenbay replied:
Done!
